gc/structs/ForwardedHeader.hpp

```diff
@@ -263,14 +347,8 @@ class MM_ForwardedHeader
 	MMINLINE void
 	commenceFixup(omrobjectptr_t destinationObjectPtr)
 	{
-		MutableHeaderFields* newHeader = (MutableHeaderFields *)((fomrobject_t *)destinationObjectPtr + _forwardingSlotOffset);
-		
-		/* copy preserved flags, and keep the rest (which should be all 0s) */
-		newHeader->slot = (_preserved.slot & ~_copyProgressInfoMask) | _beingCopiedTag;
-
-#if defined (OMR_GC_COMPRESSED_POINTERS)
-		newHeader->overlap = _preserved.overlap;
-#endif /* defined (OMR_GC_COMPRESSED_POINTERS) */
+		uintptr_t newValue = (getCanonicalPreserved() & ~_copyProgressInfoMask) | _beingCopiedTag;
```

Comment:
I'm sensing a problem here. '& ~_copyProgressInfoMask' will wipe out even overlap part, but should wipe out only upper bits of class slot.

_copyProgressInfoMask used to be 0xffffff00 in CR world (size of fomrobject_t) and now is 0xffffffffffffff00 (size of uintptr_t)

Reply:
That was exactly the issue - testing a fix now.